### PR TITLE
indicator libraries: init at 12.10

### PR DESCRIPTION
**Pull request by msteen**
The indicator libraries for GTK2 and GTK3.

The bindings for Mono do not work yet because of some issues with Perl in
gtk-sharp-2 and an Unhandled Exception caused by building with multiple cores:
https://aur.archlinux.org/packages/libindicate-sharp/#comment-290385

Some packages have TODOs for the indicator libraries,
since it is an optional dependency for most packages.
These packages have not been updated by this commit.

Besides the Mono bindings, there is still one open issue. For the packages that use autoconf a warning is given that automake-11.1 and aclocal-11.1 are missing. If the package automake-111x is added to the build inputs it will give a warning about a version mismatch between automake 11.1.5 (the package) and 11.1.6 (automake-111x), telling me to regenerate the files. However regenerating the Makefile will result in many warnings about unsupported syntax, which according to the warning were likely to be GNU extensions. Since the initial warnings are about executables that are not actually used, I chose to ignore them for now, instead of potentially breaking stuff by regenerating the Makefiles.

### CODE REVIEW

File: pkgs/development/libraries/libindicator/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    sha256 = "b2d2e44c10313d5c9cd60db455d520f80b36dc39562df079a3f29495e8f9447f";
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
+
+  buildInputs = [ (if gtkVersion == "2" then gtk2 else gtk3) ];
+
+  postPatch = ''
+    substituteInPlace configure \
+      --replace 'LIBINDICATOR_LIBS+="''$LIBM"' 'LIBINDICATOR_LIBS+=" ''$LIBM"'
```
Comment (demin-dmitriy) [inline pkgs/development/libraries/libindicator/default.nix]: Btw, there is no need in escaping single `$`. For example: `''$abc'' == ''''$abc''` and `''''${abc}'' == "\${abc}"`, but `''${abc''` is syntax error.

Comment (msteen) [inline pkgs/development/libraries/libindicator/default.nix]: You are right, it is unnecessary here. Thanks for pointing it out to me! I have updated the commit to remove the unnecessary escaping of the dollar signs.

**Comment by domenkozar** [comment]
`error: attribute ‘msteen’ missing, at /home/travis/build/NixOS/nixpkgs/pkgs/development/libraries/indicator-application/gtk2.nix:51:21`

**Comment by domenkozar** [comment]
You have to add yourself as maintainer to `lib/maintainers.nix`

**Comment by msteen** [comment]
@domenkozar I know, I did this with my first pull requests, which has not been merged yet. I just created a new pull request #10791 that only adds myself to the list of maintainers.

**Comment by domenkozar** [comment]
Sorry for confusion, github makes it hard to make PRs that depend on each other.

**Comment by msteen** [comment]
Please note that this pull request is still based on 15.09. Let me know if it needs to be rebased on master instead.

**Comment by domenkozar** [comment]
Yes, please base any work on top of `master`

**Pull request closed without merging**